### PR TITLE
[12.0][ADD] GeoEngine Bing Raster Support

### PR DESCRIPTION
**Pull request by keshrath**

### CODE REVIEW

File: geoengine_bing/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "version": "12.0.1.0.0",
+    "author": "MuK IT,Odoo Community Association (OCA)",
+    "category": "GeoBI",
+    "license": "AGPL-3",
+    "depends": [
+        "base_geoengine",
+    ],
+    "contributors": [
+        "Kerrim Abdelhamed <kerrim.abdelhamed@mukit.at>",
+        "Mathias Markl <mathias.markl @mukit.at>",
```
Comment (yvaucher) [inline geoengine_bing/__manifest__.py]:
```suggestion
        "Mathias Markl <mathias.markl@mukit.at>",
```

File: geoengine_bing/geo_view/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2019 MuK IT GmbH - Kerrim Abd El-Hamed
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import geo_raster_layer
```
Comment (yvaucher) [inline geoengine_bing/geo_view/__init__.py]: Please do insert EOL at EOF

It ease further update of the file in commit history readability and printout of files won't concat last line of file 1 with first line of file 2 (for instance with `cat` in bash)

**Review by yvaucher — CHANGES_REQUESTED** [review]
Hello,

Thanks for your contribution.

There are few things to correct to make `travis` happy:

```
======== Testing test_flake8 ========
geoengine_bing/geo_view/geo_raster_layer.py:6:1: F401 'odoo._' imported but unused
geoengine_bing/geo_view/geo_raster_layer.py:7:1: F401 'odoo.exceptions.UserError' imported but unused
geoengine_bing/geo_view/geo_raster_layer.py:11:1: E302 expected 2 blank lines, found 1
geoengine_bing/geo_view/geo_raster_layer.py:12:1: W293 blank line contains whitespace
geoengine_bing/geo_view/geo_raster_layer.py:14:1: W293 blank line contains whitespace
geoengine_bing/geo_view/geo_raster_layer.py:15:5: E265 block comment should start with '# '
geoengine_bing/geo_view/geo_raster_layer.py:15:80: E501 line too long (80 > 79 characters)
geoengine_bing/geo_view/geo_raster_layer.py:17:5: E265 block comment should start with '# '
geoengine_bing/geo_view/geo_raster_layer.py:17:80: E501 line too long (80 > 79 characters)
geoengine_bing/geo_view/geo_raster_layer.py:18:1: W293 blank line contains whitespace
geoengine_bing/geo_view/geo_raster_layer.py:21:1: W293 blank line contains whitespace
geoengine_bing/geo_view/geo_raster_layer.py:30:1: W293 blank line contains whitespace
geoengine_bing/geo_view/geo_raster_layer.py:33:5: E265 block comment should start with '# '
geoengine_bing/geo_view/geo_raster_layer.py:33:80: E501 line too long (80 > 79 characters)
geoengine_bing/geo_view/geo_raster_layer.py:35:5: E265 block comment should start with '# '
geoengine_bing/geo_view/geo_raster_layer.py:35:80: E501 line too long (80 > 79 characters)
geoengine_bing/geo_view/geo_raster_layer.py:36:1: W293 blank line contains whitespace
```

Can you fix those?

**Comment by yvaucher** [comment]
@keshrath I went on runbot to test it, as I was expecting it, it needs a key to use the maps.
As I don't have one, I can't test.

Could you:

1. put a screenshot with a view using one of the bing layer here.
2. add a mention of the need of an API key in the README usage part.

**Comment by keshrath** [comment]
> @keshrath I went on runbot to test it, as I was expecting it, it needs a key to use the maps.
> As I don't have one, I can't test.
> 
> Could you:
> 
> 1. put a screenshot with a view using one of the bing layer here.

image

> 2. add a mention of the need of an API key in the README usage part.

https://github.com/OCA/geospatial/pull/200/files#diff-d0e6f36f516b8f2a9c87080ac1455e6aR7

**Comment by keshrath** [comment]
> Hello,
> 
> Thanks for your contribution.
> 
> There are few things to correct to make `travis` happy:
> 
> ```
> ======== Testing test_flake8 ========
> geoengine_bing/geo_view/geo_raster_layer.py:6:1: F401 'odoo._' imported but unused
> geoengine_bing/geo_view/geo_raster_layer.py:7:1: F401 'odoo.exceptions.UserError' imported but unused
> geoengine_bing/geo_view/geo_raster_layer.py:11:1: E302 expected 2 blank lines, found 1
> geoengine_bing/geo_view/geo_raster_layer.py:12:1: W293 blank line contains whitespace
> geoengine_bing/geo_view/geo_raster_layer.py:14:1: W293 blank line contains whitespace
> geoengine_bing/geo_view/geo_raster_layer.py:15:5: E265 block comment should start with '# '
> geoengine_bing/geo_view/geo_raster_layer.py:15:80: E501 line too long (80 > 79 characters)
> geoengine_bing/geo_view/geo_raster_layer.py:17:5: E265 block comment should start with '# '
> geoengine_bing/geo_view/geo_raster_layer.py:17:80: E501 line too long (80 > 79 characters)
> geoengine_bing/geo_view/geo_raster_layer.py:18:1: W293 blank line contains whitespace
> geoengine_bing/geo_view/geo_raster_layer.py:21:1: W293 blank line contains whitespace
> geoengine_bing/geo_view/geo_raster_layer.py:30:1: W293 blank line contains whitespace
> geoengine_bing/geo_view/geo_raster_layer.py:33:5: E265 block comment should start with '# '
> geoengine_bing/geo_view/geo_raster_layer.py:33:80: E501 line too long (80 > 79 characters)
> geoengine_bing/geo_view/geo_raster_layer.py:35:5: E265 block comment should start with '# '
> geoengine_bing/geo_view/geo_raster_layer.py:35:80: E501 line too long (80 > 79 characters)
> geoengine_bing/geo_view/geo_raster_layer.py:36:1: W293 blank line contains whitespace
> ```
> 
> Can you fix those?

I think this should fix it: https://github.com/OCA/geospatial/commit/93658fa131870f319effe45a9c232c2506585f72

**Comment by yvaucher** [comment]
@keshrath perfect, lint check passed

**Review by yvaucher — APPROVED** [review]

**Pull request merged**